core/view/src/parser/index.ts

```diff
@@ -146,36 +77,54 @@ export class Parser {
     type: Node.ChildrenTypes = NodeType.Value,
     options: ParseObjectOptions = { templateDepth: 0 }
   ): Node.Node | null {
-    if (hasApplicability(obj)) {
-      return this.parseApplicability(obj, type, options);
+    const nodeType = this.hooks.determineNodeType.call(obj);
+
+    if (nodeType !== undefined) {
+      const parsedNode = this.hooks.parseNode.call(
+        obj,
+        type,
+        options,
+        nodeType
+      );
+      if (parsedNode) {
+        return parsedNode;
+      }
     }
 
-    if (hasSwitch(obj)) {
-      return this.parseSwitch(obj, options);
-    }
+    interface Obj {
```

Comment:
Is there a more descriptive name then `Obj` as a type we can use?

Reply:
Is there a specific reason this is defined inside of the `parseObject` method instead of a top-level interface?